### PR TITLE
Added code for Binary Tree from Inorder and Postorder traversal

### PR DESCRIPTION
**Pull request by kotwani2883**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
Added code for Binary Tree from Inorder and Postorder traversal

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#6647

### CODE REVIEW

**Comment by kotwani2883** [comment]
@siddharth25pandey  kindly review my PR . Thanks

**Review by atarax665 — APPROVED** [review]
LGTM :)

File: C-Plus-Plus/ds/BinaryTreeFromInorderPostorder.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ Enter the size of array : 8
+ Enter the inorder traversal   : 4 8 2 5 1 6 3 7
+ Enter the postorder traversal : 8 4 5 2 6 7 3 1
+ 
+ Output Format :
+ Example : ( Output to the above input example ) 
+ The Preoder Traversal of the Tree is : 1 2 4 8 5 3 6 7
+ 
+ Time/Space Complexity
+ Time Complexity : O(n^2)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/BinaryTreeFromInorderPostorder.cpp]: Time Complexity is wrong

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Small change is required, Rest is fine

**Comment by kotwani2883** [comment]
But I think the complexity is correct as we are checking the index. I am running a loop inside the Recursion so quadratic is justified in worst case when we have left/right skewed tree

**Comment by kotwani2883** [comment]
What should be the complexity according to you? and why kindly justify

**Comment by siddharth25pandey** [comment]
@kotwani2883 The Time Complexity for Binary Tree traversal is O(n) in worst case where n is the number of nodes (Incase of a Binary Search Tree). Apart from that, it only be O(h), where h is the height of the tree. (It can be O(log(n) as well in best case)
Now, coming to your point you are using a for loop for displaying the nodes elements values with traversal, Hence it's a **combination of traversal plus printing** but when we are talking about time complexity for a particular operation, you need to focus on that particular **only**.
So, either you use recursively or iteratively you will be get O(n) as worst .
Hope it makes sense.

**Comment by kotwani2883** [comment]
I am sorry I am cross questioning you againa and again but kindly look at this part of code snippet.
The traversal and recursion are separate but inside the recursion we have this for loop.
for(rootIndex=Instart ; rootIndex<=InEnd ; rootIndex++){
		if(inorder[rootIndex]==rootVal) //We have completely traversed left Subtree
		break;
	}
Since This is inside the recursion function call  So will it not make any significant effect in Time complexity.
**And in this question we have Binary Tree and not BST**

**Comment by siddharth25pandey** [comment]
@kotwani2883 the helper function you wrote is for conversion of (inorder and postorder ) to preorder traversal. This is **not same** as **traversal of tree** in any method (pre/in/post) order.
You need to write like this:
Time Complexity for performing the given operation : O(n^2)
Time Complexity for performing the traversal of the binary tree : O(n)

**Comment by kotwani2883** [comment]
@siddharth25pandey  Thank you for guiding me regarding the Time Complexity .

File: C-Plus-Plus/ds/BinaryTreeFromInorderPostorder.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+ Time Complexity for performing the given operation : O(n^2)
+ Time Complexity for performing the traversal of the binary tree : O(n)
+   Where n is the no.of nodes 
+ Space Complexity : O(1) 
+*/
+
+   Where n is the no.of nodes 
+ Space Complexity : O(1) 
+*/
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/BinaryTreeFromInorderPostorder.cpp]: Remove this

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Review by siddharth25pandey — APPROVED** [review]
@kotwani2883 LGTM 💯 , All the best for ahead contributions 👍

**Pull request merged**